Documentation/troubleshooting.rst

```diff
@@ -546,6 +546,82 @@ Opening the ``status``, we can drill down through ``policy.realized.l4``. Do
 your ``ingress`` and ``egress`` rules match what you expect? If not, the
 reference to the errant rules can be found in the ``derived-from-rules`` node.
 
+etcd (kvstore)
+==============
+
+Understanding etcd status
+-------------------------
+
+The etcd status is reported when running ``cilium status``. The following line
+represents the status of etcd:
+
+.. code:: bash
+
+   KVStore:  Ok  etcd: 1/1 connected, lease-ID=29c6732d5d580cb5, lock lease-ID=29c6732d5d580cb7, has-quorum=true: https://192.168.33.11:2379 - 3.4.9 (Leader)
+
+OK:
+  The overall status. Either ``OK`` or ``Failure``.
+
+1/1 connected:
+  Number of total etcd endpoints and how many of them are reachable.
+
+lease-ID:
+  UUID of the lease used for all keys owned by this agent
+
+lock lease-ID:
+  UUID of the lease used for locks acquired by this agent
+
+has-quorum:
+  Status of etcd quorum. Either ``true`` or set to an error.
+
+consecutive-errors:
+  Number of consecutive quorum errors. Only printed if errors are present.
+
+https://192.168.33.11:2379 - 3.4.  9 (Leader):
+  List of all etcd endpoints stating the etcd version and whether the
+  particular endpoint is currently the elected leader. If an etcd endpoint
+  cannot be reached, the error is shown.
+
+Recovery behavior
+-----------------
+
+In the event of an etcd endpoint becoming unhealthy, etcd should automatically
+resolve this by electing a new leader and by failing over to a healthy etcd
+endpoint. As long as quorum is preserved, the etcd cluster will remain
+functional.
+
+In addition, Cilium performs a background check in an interval to determine
+etcd health and potentially take action. The interval depends on the overall
+cluster size. The larger the cluster, the longer the interval:
+
+ * If no etcd endpoints can be reached, Cilium will report failure in ``cilium
+   status``. This will cause the liveness and readiness probe of Kubernetes to
+   fail and Cilium will be restarted.
+
+ * A lock is acquired and released to test a write operation which requires
+   quorum. If this operation fails, loss of quorum is reported. If quorum fails
+   for three or more intervals in a row, Cilium is declared unhealthy.
+
+ * The Cilium operator will constantly write to a heartbeat key
+   (``cilium/.heartbeat``). All Cilium agents will watch for updates to this
+   heartbeat key. This validates the ability for an agent to receive key
+   updates from etcd. If the heartbeat key is not updated in time, the quorum
+   check is declared to have failed and Cilium is declared unhealthy after 3 or
+   more consecutive failures.
```

Comment:
What are you looking for in particular? Restarts don't matter. Only if the operator is unavailable for a long period of time then complications will start to occur. That's already the case today.  

We currently have:
https://docs.cilium.io/en/v1.8/concepts/overview/#cilium

Reply:
I was thinking that the tolerance for operator being out-of-action is much higher today, depending on cluster size/activity it could be hours then starts failing in weird ways. AFAIK with this new logic we're looking at more like several minutes before Cilium starts restarting, though I think it will be more straightforward to understand/debug with this new behaviour.

(I also realize now that we'll need to consider how this impacts #12409 )

Probably we can just tweak the end of the sentence in the docs link you provided to make it more clear how operator unavailability may impact the cluster with this new logic.

Reply:
I've tweaked it further. We should consider that this PR is targeted for backports so at some point we'll have to draw a line at how many small details we touch.